### PR TITLE
Matrix Page Crashes when there is no data

### PR DESCRIPTION
**Pull request by Elkrival**
This pr fixes a bug where the matrix page crashes if no data is returned from the server. 
The issue is with certain variables calculating to numbers below 0 that cause a crash with d3.

On the graph page I moved the save button out from the Matrix div.
I made small alignment tweaks to display the numbered days better. 
There's a button that when clicked throws an alert about a feature that will be implemented in the future, this felt unnecessary to have, I removed it.

Screenshot 2024-02-19 at 3 06 42 PM
Screenshot 2024-02-19 at 3 06 54 PM

### CODE REVIEW

File: views/components/Matrix.d3.js [hunk 1]
```diff
@@ -426,15 +425,17 @@ export default class Matrix {
...
-      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1]])
+      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1 || 0]])
       .domain([0, xAxisForDatesData.length - 1])
     this.xScaleLinearBottom = d3
       .scaleLinear()
-      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1]])
-      .domain([startDay, xAxisForDatesData[xAxisForDatesData.length - 1].day])
+      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1 || 0]])
+      .domain(
+        [startDay, xAxisForDatesData[xAxisForDatesData.length - 1]?.day] || 0
```
Comment (mikestone14) [inline views/components/Matrix.d3.js:434]: should the `|| 0` go inside the array?

```suggestion
        [startDay, xAxisForDatesData[xAxisForDatesData.length - 1]?.day || 0]
```

**Review by mikestone14 — APPROVED** [review]

File: views/components/Matrix.d3.js (hunk 1 above)
Comment (Elkrival) [reply views/components/Matrix.d3.js:434]: When there is no data xAxisForDatesData is empty, if we do this `xAxisForDatesData[0]` that will result in undefined, and we need a value.

Comment (mikestone14) [reply views/components/Matrix.d3.js:434]: yeah, but don't we want the value in the array?

Comment (mikestone14) [reply views/components/Matrix.d3.js:434]: we want `[startDay, 0]` not `[startDay, undefined]` right?

Comment (Elkrival) [reply views/components/Matrix.d3.js:434]: oof, good catch it's still wrapped in the array one
👍

Comment (Elkrival) [reply views/components/Matrix.d3.js:434]: [64f55cf](https://github.com/AMP-SCZ/dpdash/pull/634/commits/64f55cf65b019cb2b9f82c6f797eb8449067a8ad)
👍

**Pull request merged**